### ISSUE TITLE
dn or dN leaves an extra newline when match is at start of line

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#101]
If you have a buffer containing
```
three
one
four
one
five
```
and say
```
/one[Enter]
dn
```
classic vi, nvi and vim leave
```
three
one
five
```
but xvi leaves
```
three

one
five
```
Similarly, with dN with the cursor on the start of the lower "one", the same effect.

This only happens when the matched pattern is at the start of a line.

**Comment by martinwguy** [comment]
Similarly, sitting on the 'o' of the first "one" and saying "d/one[Enter]" leaves a blank line
where classic vi and others don't.

If you are sitting on the "n" of the first "one" instead, xvi does the same as classic vi nvi and vim.

Similarly for yanking to a search pattern, going to the "f" of "five" and putting.
Classic vi yanking from the "o" leaves
```
five
one
four
```
but if you yanking from the "n", the `five` is replaced by:
```
fne
fourive
```
xvi in the first case changes `five` to
```
fone
fourive
```
and in the second
```
fne
fourive
```
So it looks like "Yanking or deleting from first character of a line should include the preceding newline in the yank or delete"

**Comment by martinwguy** [comment]
The "change" command has similar issues.

"c/one[Enter]X[Esc]" from the "o" it should give
```
three
X
one
five
```
and from the "n"
```
three
oX
one
five
```
Xvi gets the first one right but for the second, bizarrely, gives
```
three
Xo
one
five
```

**Comment by martinwguy** [comment]
From the results, it seems that vi performs a line-based yank/delete/change when the range starts and ends at the first character of two lines, a character-based yank/delete/change otherwise.

**Comment by martinwguy** [comment]
Fixed by 90ec56f
